common/web/keyboard-processor/src/text/systemStores.ts

```diff
@@ -42,6 +42,11 @@ namespace com.keyman.text {
     }
 
     set(value: string) {
+      // No need to signal changes when things stay the same.
+      if(this._value == value) {
+        return;
+      }
```

Comment:
While not critical, I threw this in b/c it got in the way of debugging the issue... and because it may hamper performance if/when it triggers unnecessary OSK processing.